### PR TITLE
refactor: normalize keepkey wallet state

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Noticed whilst investigating https://github.com/shapeshift/web/issues/6474 that we have two `keepKeyWallet`'s in our `KeepKeyProvider` context.

Remove the top level one and use the one that (is now) set by the reducer to match the other props.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

Medium risk to KeepKey wallet connection.

> What protocols, transaction types or contract interactions might be affected by this PR?

Interactions with the KeepKey wallet.

## Testing

Ensure KeepKey connections work as expected with no regressions. Test:

- Transactions using a KeepKey
- The KeepKey wallet menu: changing PIN, timeout, passphrase etc

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
* **#6475** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6475?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @0xApotheosis and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/context/WalletProvider/KeepKeyProvider.tsx [hunk 1]
```diff
@@ -90,14 +91,14 @@ const initialState: InitialState = {
 export interface IKeepKeyContext {
   state: InitialState
   setHasPassphrase: (enabled: boolean) => void
-  keepKeyWallet: KeepKeyHDWallet | undefined
```
Comment (0xApotheosis) [inline src/context/WalletProvider/KeepKeyProvider.tsx:93]: The crux of this PR. Remove the context `keepKeyWallet` prop and instead use the state set by the reducer.

**Review by gomesalexandre — COMMENTED** [review]
Top-level removal in favor of `state.keepKeyWallet` looks sane pending runtime testing

**Review by gomesalexandre — APPROVED** [review]
Tested locally:

- Unlock is still happy

image
image

- Disconnection is still detected

image

- Address derivation is still happy (tested with DOGE)

image

- Accounts are still happy

image

- Self-send is still happy

image

**Review by gomesalexandre — COMMENTED** [review]
Paranoia additional testing:

- Setting a passphrase is happy

image

- Using previously set passphrase is happy

image

Interestingly, the derived addresses are the exact same with/out a passphrase (i.e enabling/disabling a passphrase) but after a cache clear, the correct derived accounts show up, which is off but may not be an issue with this PR per se.

Trying to regression test it against prod is impossible as I wasn't able to test connecting with a passphrase on prod i.e https://github.com/shapeshift/web/issues/3788 and am forever in a "Waiting for passphrase" state

**Pull request merged**